### PR TITLE
Remove WC's `is_ajax` (deprecated in 6.1) in favor of proxied WP `wp_doing_ajax`

### PR DESCRIPTION
**Pull request by layoutd**
### Changes proposed in this Pull Request:

WooCommerce's `is_ajax()` function will be deprecated in version 6.1: https://github.com/woocommerce/woocommerce/blob/71cdcf322c972d51fa6c767a445652e74f1a6e90/plugins/woocommerce/includes/wc-deprecated-functions.php#L1129-L1139

This causes a deprecated warning in Google Listings & Ads:
image

This PR replaces it in `Installer` with a proxied version of `wp_doing_ajax()`, which has been included in WordPress since 4.7: https://github.com/WordPress/WordPress/blob/e6b9b961305edd32fdfacc78eb6d93b22c3c8768/wp-includes/load.php#L1487-L1503

It also swaps in the proxied `wp_doing_ajax()` version in `ActivationRedirect`.

### Detailed test instructions:
**Deprecated**
1. Install WooCommerce 6.1+ (currently `trunk` of the https://github.com/woocommerce/woocommerce repo).
2. Enable debugging (`define( 'WP_DEBUG', true );`).
3. Visit any WooCommerce page and confirm that the deprecation warning is not visible.
    - Confirms using `wp_doing_ajax` instead of `is_ajax` 

**Proxied value**
1. Reset the "redirect to onboarding" and "MC setup complete" options:
```sql
UPDATE wp_options SET option_value="yes" WHERE option_name="gla_redirect_to_onboarding";
DELETE FROM wp_options WHERE option_name="gla_mc_setup_completed_at" LIMIT 1;
```
2. Visit any admin page and confirm that you're redirected to GLA onboarding.
    - Confirms the proxied `wp_doing_ajax` correctly returns false.
    - Subsequently navigating to another page should no longer redirect to onboarding.
3. Go to the add new product page (which has regular heartbeat requests).
4. Open the network inspector panel to see all requests.
1. Reset the "redirect to onboarding" and "MC setup complete" options again:
```sql
UPDATE wp_options SET option_value="yes" WHERE option_name="gla_redirect_to_onboarding";
DELETE FROM wp_options WHERE option_name="gla_mc_setup_completed_at" LIMIT 1;
```
5. Confirm that any heartbeat or other async requests return the correct data, and aren't redirected to the onboarding URL (`wp-admin/admin.php?page=wc-admin&path=/google/start`).
    - Confirms the proxied `wp_doing_ajax` correctly returns true.

(_No changelog_)

### Changelog entry

### CODE REVIEW

File: src/Proxies/WP.php [hunk 1]
```diff
@@ -166,4 +166,15 @@ public function wp_timezone(): DateTimeZone {
 	public function number_format_i18n( float $number, int $decimals = 0 ): string {
 		return number_format_i18n( $number, $decimals );
 	}
+
+	/**
+	 * Determines whether the current request is a WordPress Ajax request.
+	 *
+	 * @return bool True if it's a WordPress Ajax request, false otherwise.
+	 *
+	 * @since x.x.x
```
Comment (puntope) [inline src/Proxies/WP.php:175]: ❓ 

x.x.x ? I guess to be replaced by 2.0.0?

Comment (layoutd) [reply src/Proxies/WP.php:175]: Yeah, that's probably right!
```suggestion
	 * @since 2.0.0
```

File: src/Admin/ActivationRedirect.php [hunk 2]
```diff
@@ -28,6 +29,19 @@ class ActivationRedirect implements Activateable, Service, Registerable, Options
...
+	protected $wp_proxy;
+
+	/**
+	 * Installer constructor.
+	 *
+	 * @param WPProxy $wp_proxy
+	 */
+	public function __construct( WPProxy $wp_proxy ) {
+		$this->wp_proxy = $wp_proxy;
+	}
```
Comment (puntope) [inline src/Admin/ActivationRedirect.php]: ❓  Why this WPProxy naming?

I saw that in 

GlobalSiteTag.php, MerchantMetrics.php, WooCommerceBrands.php ...etc. (basically all the files using this WP class) are using  it as

`$this->wp = $wp;` instead of using WPProxy naming.

File: src/Admin/BulkEdit/CouponBulkEdit.php [hunk 3]
```diff
@@ -138,6 +137,8 @@ public function handle_submission( int $post_id, $post ): int {
 	 * @return array The $_REQUEST superglobal.
 	 */
 	protected function request_data(): array {
+		// Nonce must be verified manually.
+		//phpcs:ignore WordPress.Security.NonceVerification.Recommended
```
Comment (puntope) [inline src/Admin/BulkEdit/CouponBulkEdit.php:141]: ❓ 
This chunk of code seems to be unrelated with the `is_ajax()`. Do I need to test anything here?

Comment (layoutd) [reply src/Admin/BulkEdit/CouponBulkEdit.php:141]: Nope, but it was giving a warning when I ran PHPCS so I added it to have a clean slate!

**Review by puntope — COMMENTED** [review]
✅  Behaviour looks good to me

- I don't see anymore the deprecated warning ✅ 
- I was redirected to onboarding when I run the described SQL and tried to access a page in the admin ✅ 
- In product page, with the described SQL executed, I can get right all the requests. ✅ 
- Visual code check looks good ✅ 

❓  Some questions to be addressed regarding some naming

File: src/Admin/ActivationRedirect.php (hunk 2 above)
Comment (layoutd) [reply src/Admin/ActivationRedirect.php]: 👏🏻  Good spot. I named it that because it's clearer to me what the class does, but I'll swap it back for consistency with other classes.

Comment (layoutd) [reply src/Admin/ActivationRedirect.php]: Updated in https://github.com/woocommerce/google-listings-and-ads/pull/1164/commits/7ae11d9d10d3582287bd2f68757629c8293c1c40

**Comment by layoutd** [comment]
Thanks for the thorough review, @puntope. I'll give it one more quick test to make sure these last fixes don't break anything, and merge it.

In the end I've left `@since x.x.x` (regarding [this comment](https://github.com/woocommerce/google-listings-and-ads/pull/1164#discussion_r776349082)) and hopefully we can enable auto-replace in the release process 🤞🏻

**Review by puntope — APPROVED** [review]
✅  Checked again all the cases and works 
✅  All the questions were addressed 

Ready to merge

**Comment by mikkamp** [comment]
> In the end I've left `@since x.x.x` (regarding this comment) and hopefully we can enable auto-replace in the release process 🤞🏻

We have version replacing enabled in this extension, see: https://github.com/woocommerce/google-listings-and-ads/blob/develop/package.json#L101-L107

**Comment by layoutd** [comment]
Thanks for clarifying @mikkamp, I checked in the `package.json` file and somehow missed it 😖
